gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
Thanks for the example; it's a really good one! I agree my use of direction_id is not appropriate, so let's scratch that idea.

I have a new solution that I think solves our problem; have **TWO** fields for the sequence:
- **next_station_arrival_sequence**: the order that the carriage will arrive at the next station
- **next_station_departure_sequence**: the order that the carriage will depart from the next station

For non-dead end stations, next_station_arrival_sequence = next_station_departure_sequence. For dead-end stations, they will be the reverse of each other. So for my dead-end picture above:

TUs/VPs before arriving at the station:
carriage_descriptor {
  label = 'A'
  next_station_arrival_sequence: 1
  next_station_departure_sequence: 3
},
carriage_descriptor {
  label = 'B'
  next_station_arrival_sequence: 2
  next_station_departure_sequence: 2
},
carriage_descriptor {
  label = 'C'
  next_station_arrival_sequence: 3
  next_station_departure_sequence = 1
}

Consumers can know from this that it's a dead-end station, and can decide on an appropriate UI/info to show to the user.

Thoughts?